backend/api/communication/src/main/java/co/airy/core/api/communication/service/SendMessageExecutorService.java

```diff
@@ -0,0 +1,43 @@
+package co.airy.core.api.communication.service;
+
+import co.airy.core.api.communication.Stores;
+import co.airy.core.api.communication.payload.SendMessageRequestPayload;
+import co.airy.model.conversation.Conversation;
+import org.apache.kafka.streams.state.ReadOnlyKeyValueStore;
+import org.springframework.stereotype.Component;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+@Component
+public class SendMessageExecutorService {
+
+    private static ExecutorService executor;
+
+    public Future<Conversation> getFutureCallableConversationAsync(Stores stores, SendMessageRequestPayload payload){
```

Comment:
Try to use a Thread pool. [see](https://github.com/airyhq/cloud/pull/251/files#diff-ff6863c6270f71bc344205c8ef0e836b59af63c2dfb3bb57600efe8d071029c9). Futures are expensive, specially in this case where the endpoint is going to be call often.

Reply:
Agreed. Would simply copy-paste Juan's linked solution. Bonus points for adding a small spring library in `lib/java/spring` :)